Oop№3   aggregation

LarionovKirill opened the pull request.

LarionovKirill committed:
Feature(Cart.cs): Создал класс корзины пользователя
Feature(Customer.cs):Класс композирует корзину и изменен конструктор.

LarionovKirill committed:
Feature(Customer.cs): Добавил изменения в конструкторе.
Feature(Order.cs):создал класс заказов.
Feature(Status.cs): Создал перечисление типа заказа.

LarionovKirill committed:
Merge pull request #25 from LarionovKirill/OOP-№2---Composition

Fix(Tabs): Везде постарался добавить адаптивную верстку(Если что напи…

LarionovKirill committed:
Feature(CartsTab): добавлен пользовательский интерфейс.

LarionovKirill committed:
Fix(Controls):Временное сохранение для проверки.

LarionovKirill committed:
Feature(CartsTab.cs): Исправил форму и сделал верстку.

LarionovKirill committed:
Feature(generators): Добавил генераторы товаров и пользователей.
Feature(ItemsTab.cs): добавил обновление данных при генерации.
Feature(CustomerTab.cs): Добавил обновление данных при генерации.
Feature(Carts.cs): добавил весь функционал и адаптивную верстку.

LarionovKirill committed:
Feature(OrdersTab): создал вкладку с адаптивной версткой и обновлением данных.
Fix(Customer.cs/Order.cs): исправил баги с конструкторами.
Fix(CartsTab.cs): исправил баги с версткой и кодом.

LarionovKirill committed:
Feature(OrdersTab.cs): добавил изменение типа заказа и исправил баг

LarionovKirill committed:
Fix(OrderTab.cs): исправил баг с именем.

LarionovKirill committed:
Fix(Order.cs): убрал лишнее поле.

LarionovKirill committed:
Feature(Diagrams): добавлена диаграмма классов.

LarionovKirill committed:
Fix(Uml Лаба 3.jpg): изменил диаграмму.

LarionovKirill committed:
Fix(cart.cs/item.cs/order.cs): изменил double на decimal

LarionovKirill committed:
Refactoring(): убрал лишние поля и заменил их на авто свойства.

LarionovKirill committed:
Заменил название Category на ItemCategory

Ershovoy commented on ObjectOrientedPractics/OOP/View/Tabs/CustomerTab.cs:
```diff
@@ -0,0 +1,139 @@
+        {
+            try
+            {
+                Model.Address currentAddress = GetAddress();
+                Customers.Add(new Model.Customer(
+                    fullNameTextBox.Text,
+                    currentAddress.Index,
+                    currentAddress.Building,
+                    currentAddress.City,
+                    currentAddress.Country,
+                    currentAddress.Street,
+                    currentAddress.Apartment));
```
При добавление нового покупателя вылетает исключение.
![image](https://github.com/LarionovKirill/Programming/assets/50287099/e70f8d76-8598-4ba2-9389-9114eadd39ac)

Ershovoy commented on ObjectOrientedPractics/OOP/View/Tabs/CustomerTab.cs:
```diff
@@ -0,0 +1,139 @@
+
+namespace OOP.View
+{
+    public partial class CustomerTab : UserControl
```
Изменённое имя не обновляется списке покупателей.
![image](https://github.com/LarionovKirill/Programming/assets/50287099/e0ba345e-423a-4d08-a549-65f3da9be53d)

LarionovKirill replied:
В лабе и не сказано было это делать, иначе ты бы менял имя еще и другого человека.
По-хорошему поля надо сделать ReadOnly и по кнопке давать возможность редактировать.

Ershovoy commented on ObjectOrientedPractics/OOP/View/Tabs/ItemsTab.cs:
```diff
@@ -0,0 +1,215 @@
+
+namespace OOP.View.Tabs
+{
+    public partial class ItemsTab : UserControl
```
Не редактируются данные о товаре.
![image](https://github.com/LarionovKirill/Programming/assets/50287099/2c6da4d8-2d16-40ed-853a-d161c95cb25f)

LarionovKirill committed:
Merge branch 'develop' into OOP№3---Aggregation

LarionovKirill committed:
Fix():Исправил баг из-за которого программа не запускалась
Fix(addressControe.cs / CustomerTab.cs): исправил баг с создание пользователя.
Feature(ItemsTab.cs): Добавил возможность редактирования полей по кнопкам.

LarionovKirill committed:
Временный коммит.
Feature(ItemTab.cs): добавлена возможность редактирования.
Feature(customerTab.cs): частично добавлена возможность редактирования, но надо доделать.

LarionovKirill committed:
Fix(CustomerTab.cs): доделал исправление пользователя по кнопкам.

LarionovKirill committed:
Feature(CustomerTab.cs): добавил XML

LarionovKirill committed:
.
